clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -4240,15 +4242,34 @@ scheduler_tmpl_t<RecordType, ReaderType>::print_queue_stats()
         unsched_size = unscheduled_priority_.queue.size();
     }
     int live = live_input_count_.load(std::memory_order_acquire);
-    VPRINT(this, 1, "inputs: %zd scheduleable, %zd unscheduled, %zd eof\n",
-           live - unsched_size, unsched_size, inputs_.size() - live);
+    // Make our multi-line output more atomic.
+    std::ostringstream ostr;
```

Comment:
Maybe gate the construction of the string also by the verbosity level? Or else it'll get unnecessarily expensive even for cases where we don't need to print.

Reply:
The caller is supposed to check verbosity before calling.  Otherwise the entire function should be inside a VDO; seems better to let the caller decide.  Changed the VPRINT at the bottom to 0 to not prevent someone calling at any time.  Added a comment to the .h decl.